### PR TITLE
Actions simplification

### PR DESCRIPTION
**Pull request by K-ballo**
Simplification of the actions implementation. As a result, the following macros were removed as `HPX_DEFINE_COMPONENT_[DIRECT_]ACTION` works for every case:

- `HPX_DEFINE_COMPONENT_CONST_ACTION`
- `HPX_DEFINE_COMPONENT_ACTION_TPL`
- `HPX_DEFINE_COMPONENT_CONST_ACTION_TPL`
- `HPX_DEFINE_COMPONENT_CONST_DIRECT_ACTION`
- `HPX_DEFINE_COMPONENT_DIRECT_ACTION_TPL`
- `HPX_DEFINE_COMPONENT_CONST_DIRECT_ACTION_TPL`

### CODE REVIEW

**Comment by hkaiser** [comment]
This is very nice! LGTM.  \o/

**Comment by pagrubel** [comment]
Testing to see how this affects execution times for threading experiments.

**Comment by pagrubel** [comment]
Still have errors when setting HPX_WITH_LOGGING=OFF for this branch (although it worked for master)

File: docs/CMakeLists.txt [hunk 1]
```diff
@@ -80,9 +80,9 @@ set(doxygen_dependencies
     "${hpx_SOURCE_DIR}/hpx/parallel/algorithms/uninitialized_copy.hpp"
     "${hpx_SOURCE_DIR}/hpx/parallel/algorithms/uninitialized_fill.hpp"
     "${hpx_SOURCE_DIR}/hpx/runtime/get_ptr.hpp"
-    "${hpx_SOURCE_DIR}/hpx/runtime/actions/action_support.hpp"
+    "${hpx_SOURCE_DIR}/hpx/runtime/actions/basic_action.hpp"
```
Comment (sithhell) [inline docs/CMakeLists.txt]: isn't action_support.hpp still part of the headers? Why has it been removed here?

**Comment by K-ballo** [comment]
> Still have errors when setting HPX_WITH_LOGGING=OFF for this branch (although it worked for master)

Works for me using gcc-4.9 and boost-1.56.0 (core works, some examples still fail to #include what they use on both branches). Which compiler and boost version are you using?

File: docs/CMakeLists.txt (hunk 1 above)
Comment (K-ballo) [inline docs/CMakeLists.txt]: The bits from `action_support.hpp` that contribute to the documentation have been moved to `basic_action.hpp` about two commits ago.

**Comment by sithhell** [comment]
Looks pretty good to me! Thanks for this! Please announce the breaking changes on the ML!

**Pull request merged**